### PR TITLE
Rename true/false datatype, consistent capitalization

### PR DESCRIPTION
**Pull request by callumbwhyte**
For the longest time I've found the "True/false" data type that ships with annoying...

- Most of the default data types are named after their utility, e.g. "XYZ Picker" or "Dropdown", rather than the data they store _(notable exceptions being "Numeric" and "Textstring"... 😒)_
- Whenever you save it the name changes from "True/false" to "Truefalse" – GAH!
- The property editor alias is `Umbraco.TrueFalse`, the icon is a checkbox, but pre values all refer to it as a toggle...

image

This PR renames the default "True/false" data type to "Toggle" for new installs... This also aligns with the current naming in the new backoffice

I have also adjusted the casing of some of the other default data type names for consistency...

### CODE REVIEW

**Comment by JasonElkin** [comment]
Ahoy @callumbwhyte, thanks for this.

In testing this throws up a couple of inconsistencies that I think should be looked at, or at least talked about, as part of this PR.

Firstly there's inconsistent casing/naming conventions in use below that I think should be updated to title case and renamed to better match the data types:

image

[Docs](https://docs.umbraco.com/umbraco-cms/fundamentals/data/data-types/default-data-types) will need to be amended to reflect the change. I don't know how minor releases are reflected in docs - I'll check.

I don't know what the devil "Radiobox" is supposed to mean...📻..?

I'm also not sure why the icon for "Date Picker" is a clock. when `icon-calendar-alt` exists...

image

File: src/Umbraco.Infrastructure/Migrations/Install/DatabaseDataCreator.cs [hunk 1]
```diff
@@ -539,7 +539,7 @@ void InsertDataTypeNodeDto(int id, int sortOrder, string uniqueId, string text)
                 Path = $"-1,{Constants.DataTypes.DateTime}",
                 SortOrder = 2,
                 UniqueId = Constants.DataTypes.Guids.DatePickerWithTimeGuid,
-                Text = "Date Picker with time",
+                Text = "Date Picker With Time",
```
Comment (JasonElkin) [inline src/Umbraco.Infrastructure/Migrations/Install/DatabaseDataCreator.cs:542]: English language style guides disagree as to whether four-letter propositions should be capitalised in title case. So either "Date Picker with Time" or "Date Picker With Time" could be valid here.

It's time to be opinionated. I prefer the former, which anecdotally seems more common. Other opinions welcome.

```suggestion
                Text = "Date Picker with Time",
```

Comment (JasonElkin) [reply src/Umbraco.Infrastructure/Migrations/Install/DatabaseDataCreator.cs:542]: Tagging @nathanwoulfe, thanks for making this easier 😅
```suggestion
                Text = "Date and Time Picker",
```

**Comment by marcemarc** [comment]
If we are going crazy open season on these names.... how about 'Repeatable textstrings' it's repeating a singular item so I don't think should be pluralised, and the thing being repeated is a Textbox ? so would  'Repeatable Textbox' be more descriptive?

**Comment by marcemarc** [comment]
If Richtext editor is becoming Richtext Editor, then should Markdown editor do the same?

Sacrilege to suggest it, but Multinode Treepicker is it just Treepicker?

Is Multi Url Picker just 'Url Picker'?

Is Eye Dropper Color Picker just 'Color Picker'?

Then with the 'Lists'

Checkbox List
Dropdown List
Radio button List

??

**Comment by ronaldbarendse** [comment]
I think this will quickly become a tabs/spaces type of discussion and whatever names we're updating to will annoy someone else again 😇

A valid issue though is the default `True/false` data type name that gets renamed when saved, so that should indeed get fixed (and `Toggle` seems to best name for this editor indeed) 👍🏻 Don't forget to create a PR to [update the documentation](https://docs.umbraco.com/umbraco-cms/fundamentals/data/data-types/default-data-types#true-false) as well 😉

One major thing to note however, is that changing the default data type names might cause issues on installs that expect them to exist, like when using Konstrukt and [setting the data type using the name](https://docs.getkonstrukt.net/collections/editors#changing-the-data-type-of-a-field). Even changing the casing might cause issues, because the default `IDataTypeService.GetDataType(string name)` implementation uses an exact match in the WHERE query (although your database could be configured to use case-insensitive comparisons, depending on the SQL Server collation):

https://github.com/umbraco/Umbraco-CMS/blob/fe2d071fd559b0420cc469f7dec99816c6e89090/src/Umbraco.Core/Services/DataTypeService.cs#L307-L318

We can avoid most data type naming issues if Umbraco didn't create them by default during install, which you can currently already do by configuring the [`InstallDefaultData` settings](https://docs.umbraco.com/umbraco-cms/reference/configuration/installdefaultdatasettings). I don't see any value in creating an empty dropdown or checkbox radiobox list, since you'd have to add prevalues/options to them before being useful anyway.

**Comment by c9mb** [comment]
@ronaldbarendse Indeed, toggle is a better name, and is in line with other framework components, such as Vue, which refer to it as either a switch or toggle.

**Comment by nul800sebastiaan** [comment]
Hi there @callumbwhyte  and thanks for the work on this one.

Looking at the impact of renaming things now and the discussions it's already starting, and the docs updates needed, we suggest people with strong opinions to start a discussion with proposals for the new backoffice (Bellissima) instead.

**Pull request closed without merging**